### ISSUE TITLE
Confusion about permissions and expected/intended file/folder owners

### ISSUE THREAD

**Issue by BhaaLseN** [WeblateOrg/weblate#2418]
Since 3.2 and the offloading of tasks to Celery, it seems we've accumulated a bunch of inconsistent (and as seen in #2400 troublesome) permissions on various files and folders; mostly in the `DATA_DIR`.
I've checked the docs, but there wasn't any obvious page that indicates what the intended owners (or groups; and as extension to that, permissions) for the various files and folders should be.

Files in `DATA_DIR/whoosh` were owned by both `www-data` (the Apache2 web server user which is serving Weblate using WSGI) and `weblate` (the local user that is used to clone/fetch the Git copy, as well as the one that is used for the Celery workers and is specified in `WSGIDaemonProcess` as user).
In #2400, Celery (running as `weblate`) failed due to files being owned by `www-data` and chown'ing them to `weblate` fixed the issue.

However, then one of the repositories failed to push successfully (both using the `commitgit` cronjob running from the `www-data` user, which is likely also wrong; and the Repository Maintenance "Push" button, which _should_ be running as `weblate` over WSGI by my limited understanding), after being a little overzealous and chown'ing all of `DATA_DIR`. Pushing failed due to `DATA_DIR/ssh/id_rsa` not being readable by the user, and it only worked after chown'ing it back from `weblate` to `www-data`.

With that, I'm really confused about which user it should be (gut feeling: should be `weblate` everywhere).
And moreover, I'm worried that some other permissions might be screwed that break silently somewhere in the back; or at least in some log file that I don't know of (and thus don't read regularly to check for errors).

**Comment by nijel** [comment]
All services should be executed under same user and that one should own all files under `DATA_DIR`. In upcoming 3.3 release this will be checked at runtime (see  e1b0c87da6ff3a782619b06f0d3e41062cabc71f and 13603d5e2ec9414884140093ce81e1be76f581b6).

**Comment by BhaaLseN** [comment]
Looks like that could've been the missing piece. Thanks!

I'm guessing the cronjobs should _also_ run under that user? I've had issues running them under that user and moved them to `www-data` when I installed the machine over a year ago; but I don't exactly remember what problem it was exactly.

**Comment by nijel** [comment]
Yes, but cron jobs are not needed in default setup and thus  they are not covered by the docs.

**Comment by Johnson145** [comment]
I'm still confused about the different users. When I originally installed Weblate there wasn't any _weblate_ user and so there isn't one yet either. The `weblate.ini` file didn't used the uid and gid settings. The `DATA_DIR` was owned by `www-data`.

What steps do I now need to do to get the 3.2.2 version running correctly (currently I can't start celery as a daemon)? If I just create a new user called `weblate` and change the ownership of the `DATA_DIR` to that user, at least nginx will complain about missing permissions. Adding the `uid` and `gid` settings to the `weblate.ini` file doesn't fix the problem in my case. Do I need to do anything else? Are there any additional/different steps for the case that I've installed Weblate using pip?

Btw, the current documentation doesn't mention the creation of the weblate user yet, does it?

**Comment by nijel** [comment]
Just run Celery as `www-data`, that's fine as well. It just needs to be same as the one running uwsgi.
